Add SDK specificity across the docs for the Customizing Clerk section

SarahSoutoul opened the pull request:
### 🔎 Previews:

- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/appearance-prop/overview
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/appearance-prop/layout
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/appearance-prop/themes 
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/appearance-prop/variables 
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/appearance-prop/captcha
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/adding-items/organization-profile
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/adding-items/user-profile 
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/adding-items/user-button 
- [x] https://clerk.com/docs/pr/ss-sdk-specificity-project-customizing-clerk/guides/customizing-clerk/localization

### What does this solve?

This PR is part of the overall SDK-specificity project, but specifically targets the pages within the Customizing Clerk section. The pages that include code examples are listed above for testing. 

### What changed?

- Added code examples for all necessary SDKs, using either <If> components or <Tabs> depending on the use case. 
- Removed Remix code examples.

**Note:** This could be a great place to use a `<If not={["Astro", "JavaScript", "Vue", "Nuxt", "Fastify"]}>` as mentioned by @alexisintech, but until then, this is my suggestion. 

### Checklist

- [ ] I have clicked on "Files changed" and performed a thorough self-review
- [ ] All existing checks pass

alexisintech commented on docs/guides/customizing-clerk/appearance-prop/layout.mdx:
```diff
@@ -5,7 +5,7 @@ description: Utilize Clerk's layout prop in order to change the layout of the <S
 {/* JS file: https://github.com/clerk/javascript/blob/main/packages/types/src/appearance.ts#L538 */}
 
-The `layout` property can be used to change the layout of the [`<SignIn/>`](/docs/reference/components/authentication/sign-in) and [`<SignUp/>`](/docs/reference/components/authentication/sign-up) components, as well as set important links to your support, terms, and privacy pages.
+The `layout` property can be used to change the layout of the [`<SignIn/>`](/docs/reference/components/authentication/sign-in) and [`<SignUp/>`](/docs/reference/components/authentication/sign-up) components, as well as set important links to your support, terms, and privacy pages. It is passed as a parameter to the [`appearance` prop](/docs/guides/customizing-clerk/appearance-prop/overview).
```
ooo i like this addition!

SarahSoutoul commented on docs/guides/customizing-clerk/appearance-prop/captcha.mdx:
```diff
@@ -36,27 +36,29 @@ The `captcha` property can be used to change the appearance of the CAPTCHA widge
 ## Usage
 
-<Tabs items={["Next.js", "Astro", "React", "Remix", "Nuxt", "Vue"]}>
+<Tabs items={["SDKs using <ClerkProvider/>", "Astro", "JavaScript", "Vue", "Nuxt", "Fastify"]}>
```
Okay have just pushed changes with a structure that works better I think. Let me know thoughts @alexisintech.

[Restructure the code examples and wordign](https://github.com/clerk/clerk-docs/pull/2746/commits/96df89fd9330fd468de858ee0b149ec44d3b0b2e)

manovotny commented on docs/guides/customizing-clerk/adding-items/organization-profile.mdx:
```diff
@@ -50,9 +50,9 @@ To add a custom page to the `<OrganizationProfile />` component, use the `<Organ
+<Tabs items={["Next.js", "Astro", "JavaScript", "Vue"]}>
   <Tab>
-    The following example demonstrates two ways that you can render content in a custom page: as a component or as a direct child.
+    The following example demonstrates two ways that you can render content in a custom page: as a component or as a direct child. This is written for Next.js App Router but can be adapted for any React-based SDK.
```
Should the `This is written for Next.js App Router but can be adapted for any React-based SDK.` text into a partial? It's used 15 times in this PR. 

It may not be worth it. Just caught my eye.

SarahSoutoul replied:
Great point. Done here: [Add partial + reframe wording to use partial](https://github.com/clerk/clerk-docs/pull/2746/commits/593f581cfed0cfd40df066944a8ca4e9340e4ea7).

manovotny reviewed (approved):
I would like to @alexisintech to review, but this looks good to me.

alexisintech commented:
This is looking perfect Sarah, awesome work here 😸💖

I still feel very strongly that we should get a `not` prop - because what happens if we want to support a new SDK. we need to figure out where these lists are, and add it, and this implementation feels backwards/unintuitive

I'd like to consider this PR blocked until we have it

@NWylynko

SarahSoutoul commented:
> This is looking perfect Sarah, awesome work here 😸💖
> 
> I still feel very strongly that we should get a `not` prop - because what happens if we want to support a new SDK. we need to figure out where these lists are, and add it, and this implementation feels backwards/unintuitive
> 
> I'd like to consider this PR blocked until we have it
> 
> @NWylynko

Have now reframed everything [here](https://github.com/clerk/clerk-docs/pull/2746/commits/8c2408c6e0ae64e59dfc462f384492d3a2f315a9) with the addition of the `:not` prop. Merged [2804](https://github.com/clerk/clerk-docs/pull/2804) into this PR for things to work. It'll be good to get some eyes on the changes after adding this new prop, so we can be ready to merge this as soon as 2804 is live! @alexisintech @NWylynko @manovotny

alexisintech commented on docs/guides/customizing-clerk/appearance-prop/overview.mdx:
````diff
@@ -138,145 +138,202 @@ Remove the `cl-` prefix from a class and use it as the key for a new object in t
-```tsx {{ mark: [4] }}
-<SignIn
-  appearance={{
+<If notSdk={["vue", "nuxt"]}>
````
beautiful use of the notSdk prop! 😸💖

alexisintech commented on docs/guides/customizing-clerk/appearance-prop/themes.mdx:
```diff
@@ -743,153 +590,67 @@ In the following example, the primary color of the themes are customized.
-    }
-    const clerkPubKey = process.env.REACT_APP_CLERK_PUBLISHABLE_KEY
+<If notSdk={["astro", "js-frontend", "vue", "nuxt", "fastify"]}>
+  For React-based SDKs, pass the `variables` property to the `appearance` prop of the [`<ClerkProvider>`](/docs/reference/components/clerk-provider) component.
```
they can pass these variables to anywhere that accepts the `appearance` prop - which is not only the clerkprovider, but any component. could you remove this from the other examples as well - i won't bloat the GH comments

```suggestion
```

alexisintech commented on docs/guides/customizing-clerk/appearance-prop/themes.mdx:
```diff
@@ -92,647 +92,494 @@ To use the simple theme, set `theme` to `simple`:
-      )
-    }
+<If notSdk={["astro", "js-frontend", "vue", "nuxt", "fastify"]}>
+  For React-based SDKs, pass the `theme` property to the `appearance` prop of the [`<ClerkProvider>`](/docs/reference/components/clerk-provider) component.
```
this section is about stacking multiple themes - they could do this either on the clerk integration, or on a component itself.

i wouldn't give these instructions, as its selling the idea that the appearance prop is only passed to clerkprovider (but its also passed to component singletons)

i'd probably just remove these explanation paragraphs

SarahSoutoul replied:
Changes applied: [Refinements post feedback - part I](https://github.com/clerk/clerk-docs/pull/2746/commits/5ceae1b835f409e76db9659b19c26d68f49fb6bd)

alexisintech reviewed (commented):
some structural issues, and it seems some pages may be missing sdk-specific examples (still only showing nextjs ones)

let me know if you want to hop on a call to discuss anything!

SarahSoutoul commented on docs/guides/customizing-clerk/appearance-prop/themes.mdx:
```diff
@@ -92,647 +92,494 @@ To use the simple theme, set `theme` to `simple`:
 ### Apply a theme to all Clerk components
 
-To apply a theme to all Clerk components, pass the `appearance` prop to the [`<ClerkProvider>`](/docs/reference/components/clerk-provider) component. The `appearance` prop accepts the property `theme`, which can be set to a theme.
+The `theme` property is used to apply a theme to all Clerk components.
```
I hear you. Much better reframing here. Gonna apply these suggestions to a lot of these pages. @alexisintech

SarahSoutoul commented on docs/guides/customizing-clerk/appearance-prop/captcha.mdx:
````diff
@@ -36,113 +38,59 @@ The `captcha` property can be used to change the appearance of the CAPTCHA widge
-  <Tab>
-    ```tsx {{ prettier: false, filename: 'app.tsx' }}
-    import { ClerkProvider } from '@clerk/nextjs';
+<If notSdk={["astro", "js-frontend", "vue", "nuxt", "fastify"]}>
````
I'm not sure about iOS and Android tbh. @swolfand @mikepitre any ideas? To my understanding, iOS has its own customization system with [ClerkTheme](https://clerk.com/docs/reference/ios/clerk-theme). And Android, I'm not sure.

SarahSoutoul replied:
Yup that's the one I linked above. Thanks @mikepitre. 

Two options I can see here then @alexisintech:
1. Add an `<If> `block for iOS and android that shows a paragraph stating these SDKs have their own customization system linking to that ClerkTheme page
2. Scope the page to everything but android and iOS. But I'd need to do this for all of the customizing-clerk pages I think. 

Let me know what you think.

SarahSoutoul commented on docs/guides/customizing-clerk/appearance-prop/themes.mdx:
```diff
@@ -92,647 +92,494 @@ To use the simple theme, set `theme` to `simple`:
-    import './App.css'
-    import { dark, neobrutalism } from '@clerk/themes'
-    import { ClerkProvider } from '@clerk/clerk-react'
+#### Example
```
Changes applied: [Refinements post feedback - part I](https://github.com/clerk/clerk-docs/pull/2746/commits/5ceae1b835f409e76db9659b19c26d68f49fb6bd)

alexisintech commented on docs/guides/customizing-clerk/adding-items/organization-profile.mdx:
```diff
@@ -63,118 +66,133 @@ To add a custom page to the `<OrganizationProfile />` component, you will need t
-<If notSdk={["astro", "js-frontend", "vue"]}>
-  The following example demonstrates two ways that you can render content in a custom page: as a component or as a direct child.
+<If notSdk="js-frontend">
+  The following example demonstrates two ways that you can render content in a custom page: **as a component** or **as a direct child**.
```
i don't think this part is true for astro, vue, nuxt - it looks like those are just showing how to render content as a direct child

alexisintech commented on docs/guides/customizing-clerk/adding-items/organization-profile.mdx:
```diff
@@ -432,125 +493,110 @@ To add an external link to the `<OrganizationProfile />` navigation sidenav, you
 
-<If notSdk={["astro", "vue"]}>
-  <Include src="_partials/nextjs/nextjs-app-router-preface" />
+The following example adds a custom page as the first item in the sidenav, followed by a custom link to the homepage, and then the default members and general pages.
```
```suggestion
The following example adds a custom page as the first item in the sidenav, followed by a custom link to the homepage, and then the default `Members` and `General` pages.
```

SarahSoutoul commented on docs/guides/customizing-clerk/adding-items/user-profile.mdx:
```diff
@@ -63,36 +66,30 @@ To add a custom page to the `<UserProfile />` component, you will need to use on
-
-    import { UserButton } from '@clerk/nextjs'
+<If notSdk="js-frontend">
+  The following example demonstrates two ways that you can render content in a custom page: **as a component** or **as a direct child**.
```
Done here: [Final changes post feedback](https://github.com/clerk/clerk-docs/pull/2746/commits/06bd423633a27616c5a772454f00834337f52ca2). @alexisintech

SarahSoutoul commented on docs/guides/customizing-clerk/adding-items/user-button.mdx:
````diff
@@ -522,20 +577,38 @@ The following example adds a "Create organization" link to the `<UserButton />`
-  import { UserButton } from '@clerk/nextjs'
-
+<If notSdk={["astro", "js-frontend", "vue", "nuxt"]}>
+  ```tsx {{ filename: 'page.tsx' }}
````
Done here: [Final changes post feedback](https://github.com/clerk/clerk-docs/pull/2746/commits/06bd423633a27616c5a772454f00834337f52ca2). @alexisintech

SarahSoutoul commented on docs/_partials/nextjs/nextjs-app-router-preface.mdx:
```diff
@@ -0,0 +1 @@
+This is written for Next.js App Router but can be adapted for any React-based SDK.
```
Oh thought I had removed it.

alexisintech commented:
> @alexisintech Have submitted all changes from your comments/ feedback if you wanted to have a look. Last remaining thing is addressing this [comment](https://github.com/clerk/clerk-docs/pull/2746#discussion_r2579234994) - will do this today.

just re-reviewed everything and I think everythings looking perfect! just that last comment and the merge conflicts

SarahSoutoul commented:
@alexisintech okay here is the commit to address this [comment](https://github.com/clerk/clerk-docs/pull/2746#discussion_r2579234994): [Scope pages to exclude ios and android](https://github.com/clerk/clerk-docs/pull/2746/commits/9720ad1994f1605e67a31acd64e7d081af6af00a). 

I've only scoped the pages under "Component customization" to **only exclude ios** since ClerkTheme isn't available on Android yet. In the scoping of these pages, I've added all SDKs, even express, js-backend, for e.g., as I don't think we want to hide these pages. But happy to be told otherwise. 

Finally, I've moved ClerkTheme to be under `guides/customizing-clerk` instead, and added a redirect,  and added it to the `manifest.json `scoping it to `ios`.
